OpenRA.Mods.RA/Traits/Disguise.cs

```diff
@@ -48,10 +48,16 @@ public Player Owner
 		{
 			get
 			{
-				if (disguise.Disguised)
-					return self.Owner == self.World.LocalPlayer ? self.Owner : disguise.AsPlayer;
+				if (!disguise.Disguised)
+					return self.Owner;
 
-				return self.Owner;
+				Player p = null;
+				if (self.World.LocalPlayer != null && self.World.LocalPlayer.WinState == WinState.Undefined)
```

Comment:
@deniz1a, our misunderstanding comes from the fact that stances are not a symmetric relation. `Everyone` considers every player an ally, but all players consider `Everyone` as neutral. So when making this check, you have to pay attention whether you call `self.Owner.IsAlliedWith` or `RenderPlayer.IsAlliedWith` because they will return different values.
You can either use `RenderPlayer.InternalName == "Everyone" || self.Owner.IsAlliedWith(RenderPlayer)` or `RenderPlayer != null && RenderPlayer.IsAlliedWith(self.Owner)`.